### PR TITLE
adding lerp norm and remap

### PR DESCRIPTION
**Pull request by typemytype**
fixing #414 

how should this be documented?
now non of the additional namespace callbacks are documented

### CODE REVIEW

File: drawBot/context/tools/drawBotbuiltins.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+def norm(value, start, stop):
+    """
+    Return the interpolation factor (between 0 and 1) of a VALUE between START and STOP.
+    """
+    return float(value - start) / float(stop - start)
+
+
+def lerp(start, stop, factor):
+    """
+    Interpolate using a value between 0 and 1
```
Comment (justvanrossum) [inline drawBot/context/tools/drawBotbuiltins.py:11]:
```suggestion
    Return an interpolated value between `start` and `stop` using interpolation factor `factor`.
```

File: drawBot/context/tools/drawBotbuiltins.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
+def norm(value, start, stop):
+    """
+    Return the interpolation factor (between 0 and 1) of a VALUE between START and STOP.
```
Comment (justvanrossum) [inline drawBot/context/tools/drawBotbuiltins.py:4]:
```suggestion
    Return the interpolation factor (between 0 and 1) of a `value` between `start` and `stop`.
```

File: drawBot/context/tools/drawBotbuiltins.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+def remap(value, start1, stop1, start2, stop2, clamp=False):
+    """
+    Re-maps a number from one range to another.
+    """
+    factor = norm(value, start1, stop1)
+    if clamp:
+        if factor < 0:
+            factor = 0
+        if factor > 1:
+            factor = 1
```
Comment (justvanrossum) [inline drawBot/context/tools/drawBotbuiltins.py:25]:
```python
    if clamp:
        factor = min(1, max(factor, 0))
```

File: tests/testMisc.py [hunk 4]
```diff
@@ -342,6 +342,25 @@ def test_ttc_IndexError(self):
...
+        assert drawBot.lerp(10, 30, 0.5) == 20
+        assert drawBot.lerp(10, 10, 0.5) == 10
+        assert drawBot.lerp(10, 30, 0) == 10
+        assert drawBot.lerp(10, 30, 1) == 30
+        assert drawBot.lerp(10, 30, 2) == 50
+
+    def test_remap(self):
+        assert drawBot.remap(15, 10, 20, 30, 50) == 40
+        with self.assertRaises(ZeroDivisionError):
+            drawBot.remap(15, 20, 20, 30, 50)
```
Comment (justvanrossum) [inline tests/testMisc.py:361]: add a clamping test as `test_remap_clamp`?

File: drawBot/context/tools/drawBotbuiltins.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+
+def norm(value, start, stop):
+    """
+    Return the interpolation factor (between 0 and 1) of a VALUE between START and STOP.
+    """
+    return float(value - start) / float(stop - start)
```
Comment (justvanrossum) [inline drawBot/context/tools/drawBotbuiltins.py:6]:
```suggestion
    return (value - start) / (stop - start)
```

File: drawBot/context/tools/drawBotbuiltins.py (hunk 3 above)
Comment (typemytype) [reply drawBot/context/tools/drawBotbuiltins.py:25]: fine by me, I just find it less readable :)

Comment (justvanrossum) [reply drawBot/context/tools/drawBotbuiltins.py]: Fine with me, too, but then make the second if an elif.

Comment (typemytype) [reply drawBot/context/tools/drawBotbuiltins.py:25]: jep, changed it back

**Review by justvanrossum — APPROVED** [review]

**Pull request merged**